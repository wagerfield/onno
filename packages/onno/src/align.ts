import * as C from "csstype"
import * as T from "./types"
import { style } from "./style"

// Align Content

export type AlignContentValue = C.AlignContentProperty

export type AlignContentProp = T.Prop<AlignContentValue>

export interface AlignContentProps extends T.ThemeProps {
  alignContent?: AlignContentProp
  alc?: AlignContentProp
}

export interface AlignContentStyle extends T.Style {
  alignContent?: AlignContentValue
}

export const alignContent = style<AlignContentProps, AlignContentStyle>({
  propsKeys: ["alignContent", "alc"]
})

// Align Items

export type AlignItemsValue = C.AlignItemsProperty

export type AlignItemsProp = T.Prop<AlignItemsValue>

export interface AlignItemsProps extends T.ThemeProps {
  alignItems?: AlignItemsProp
  ali?: AlignItemsProp
}

export interface AlignItemsStyle extends T.Style {
  alignItems?: AlignItemsValue
}

export const alignItems = style<AlignItemsProps, AlignItemsStyle>({
  propsKeys: ["alignItems", "ali"]
})

// Align Self

export type AlignSelfValue = C.AlignSelfProperty

export type AlignSelfProp = T.Prop<AlignSelfValue>

export interface AlignSelfProps extends T.ThemeProps {
  alignSelf?: AlignSelfProp
  als?: AlignSelfProp
}

export interface AlignSelfStyle extends T.Style {
  alignSelf?: AlignSelfValue
}

export const alignSelf = style<AlignSelfProps, AlignSelfStyle>({
  propsKeys: ["alignSelf", "als"]
})

// Justify Content

export type JustifyContentValue = C.JustifyContentProperty

export type JustifyContentProp = T.Prop<JustifyContentValue>

export interface JustifyContentProps extends T.ThemeProps {
  justifyContent?: JustifyContentProp
  jfc?: JustifyContentProp
}

export interface JustifyContentStyle extends T.Style {
  justifyContent?: JustifyContentValue
}

export const justifyContent = style<JustifyContentProps, JustifyContentStyle>({
  propsKeys: ["justifyContent", "jfc"]
})

// Justify Items

export type JustifyItemsValue = C.JustifyItemsProperty

export type JustifyItemsProp = T.Prop<JustifyItemsValue>

export interface JustifyItemsProps extends T.ThemeProps {
  justifyItems?: JustifyItemsProp
  jfi?: JustifyItemsProp
}

export interface JustifyItemsStyle extends T.Style {
  justifyItems?: JustifyItemsValue
}

export const justifyItems = style<JustifyItemsProps, JustifyItemsStyle>({
  propsKeys: ["justifyItems", "jfi"]
})

// Justify Self

export type JustifySelfValue = C.JustifySelfProperty

export type JustifySelfProp = T.Prop<JustifySelfValue>

export interface JustifySelfProps extends T.ThemeProps {
  justifySelf?: JustifySelfProp
  jfs?: JustifySelfProp
}

export interface JustifySelfStyle extends T.Style {
  justifySelf?: JustifySelfValue
}

export const justifySelf = style<JustifySelfProps, JustifySelfStyle>({
  propsKeys: ["justifySelf", "jfs"]
})

// Place Content

export type PlaceContentValue = C.PlaceContentProperty

export type PlaceContentProp = T.Prop<PlaceContentValue>

export interface PlaceContentProps extends T.ThemeProps {
  placeContent?: PlaceContentProp
  plc?: PlaceContentProp
}

export interface PlaceContentStyle extends T.Style {
  placeContent?: PlaceContentValue
}

export const placeContent = style<PlaceContentProps, PlaceContentStyle>({
  propsKeys: ["placeContent", "plc"]
})

// Place Items

export type PlaceItemsValue = C.PlaceItemsProperty

export type PlaceItemsProp = T.Prop<PlaceItemsValue>

export interface PlaceItemsProps extends T.ThemeProps {
  placeItems?: PlaceItemsProp
  pli?: PlaceItemsProp
}

export interface PlaceItemsStyle extends T.Style {
  placeItems?: PlaceItemsValue
}

export const placeItems = style<PlaceItemsProps, PlaceItemsStyle>({
  propsKeys: ["placeItems", "pli"]
})

// Place Self

export type PlaceSelfValue = C.PlaceSelfProperty

export type PlaceSelfProp = T.Prop<PlaceSelfValue>

export interface PlaceSelfProps extends T.ThemeProps {
  placeSelf?: PlaceSelfProp
  pls?: PlaceSelfProp
}

export interface PlaceSelfStyle extends T.Style {
  placeSelf?: PlaceSelfValue
}

export const placeSelf = style<PlaceSelfProps, PlaceSelfStyle>({
  propsKeys: ["placeSelf", "pls"]
})

// Order

export type OrderValue = C.Globals | number

export type OrderProp = T.Prop<OrderValue>

export interface OrderProps extends T.ThemeProps {
  order?: OrderProp
  ord?: OrderProp
}

export interface OrderStyle extends T.Style {
  order?: OrderValue
}

export const order = style<OrderProps, OrderStyle>({
  propsKeys: ["order", "ord"]
})
