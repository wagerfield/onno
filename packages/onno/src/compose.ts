import * as T from "./types"

const KEYS: T.StyleOptionsKeys[] = ["propsKeys", "styleKeys", "themeKeys"]

const push = Array.prototype.push

export function unique(
  renderers: T.AnyRenderFunction[],
  initial: T.AnyRenderFunction[] = []
): T.AnyRenderFunction[] {
  return renderers.reduce((collection, renderer) => {
    if (renderer.options.renderers) {
      unique(renderer.options.renderers, initial)
    } else if (collection.indexOf(renderer) === -1) {
      collection.push(renderer)
    }
    return collection
  }, initial)
}

export function compose<P extends T.ThemeProps, S extends T.Style>(
  options: T.ComposeOptions
): T.ComposedRenderFunction<P, S> {
  let { name } = options
  if (!/Set$/.test(name)) name += "Set"
  const renderers = unique(options.renderers)
  const composedOptions: T.ComposedRenderOptions = {
    propsKeys: [],
    styleKeys: [],
    themeKeys: [],
    renderers
  }

  // Build options keys
  renderers.forEach((fn) =>
    KEYS.forEach((key) => {
      const keys = fn.options[key]
      if (keys) push.apply(composedOptions[key], keys)
    })
  )

  // Create scoped renderProps style function
  const renderProps: T.ComposedRenderFunction<P, S> = (props: P) => {
    const result: T.StyleArray<S> = renderers.reduce((styles, renderer) => {
      const output = renderer(props)
      if (output) push.apply(styles, output)
      return styles
    }, [])

    // Return the composed style object
    return result.length ? result : null
  }

  // Define renderProps function properties
  renderProps.options = composedOptions
  renderProps.type = "compose" as "compose"
  Object.defineProperty(renderProps, "name", { value: name })

  // Return renderProps function
  return renderProps
}
