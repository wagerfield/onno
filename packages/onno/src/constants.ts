import * as T from "./types"

export const PX_SCALE: T.ThemeValue = [0, 2, 4, 8, 16, 32, 64, 128, 256, 512]

export const PC_SCALE: T.ThemeValue = Object.assign([], PX_SCALE, { 1: "100%" })

export const BREAKPOINTS: T.Breakpoints = ["all", "sm", "md", "lg", "xl"].map(
  (alias, index) => ({ alias, value: index * 360 })
)
