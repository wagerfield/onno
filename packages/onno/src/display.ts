import * as C from "csstype"
import * as T from "./types"
import { compose, style } from "./style"

// Display

export type DisplayValue = C.DisplayProperty

export type DisplayProp = T.Prop<DisplayValue>

export interface DisplayProps extends T.ThemeProps {
  display?: DisplayProp
  d?: DisplayProp
}

export interface DisplayStyle extends T.Style {
  display?: DisplayValue
}

export const display = style<DisplayProps, DisplayStyle>({
  propsKeys: ["display", "d"]
})

// Opacity

export type OpacityValue = number | string

export type OpacityProp = T.Prop<OpacityValue>

export interface OpacityProps extends T.ThemeProps {
  opacity?: OpacityProp
  o?: OpacityProp
}

export interface OpacityStyle extends T.Style {
  opacity?: OpacityValue
}

export const opacity = style<OpacityProps, OpacityStyle>({
  propsKeys: ["opacity", "o"],
  themeKeys: ["opacities"]
})

// Overflow

export type OverflowValue = C.OverflowProperty

export type OverflowProp = T.Prop<OverflowValue>

export interface OverflowProps extends T.ThemeProps {
  overflow?: OverflowProp
  of?: OverflowProp
}

export interface OverflowStyle extends T.Style {
  overflow?: OverflowValue
}

export const overflow = style<OverflowProps, OverflowStyle>({
  propsKeys: ["overflow", "of"]
})

// Overflow X

export type OverflowXValue = C.OverflowXProperty

export type OverflowXProp = T.Prop<OverflowXValue>

export interface OverflowXProps extends T.ThemeProps {
  overflowX?: OverflowXProp
  ofx?: OverflowXProp
}

export interface OverflowXStyle extends T.Style {
  overflowX?: OverflowXValue
}

export const overflowX = style<OverflowXProps, OverflowXStyle>({
  propsKeys: ["overflowX", "ofx"]
})

// Overflow Y

export type OverflowYValue = C.OverflowYProperty

export type OverflowYProp = T.Prop<OverflowYValue>

export interface OverflowYProps extends T.ThemeProps {
  overflowY?: OverflowYProp
  ofy?: OverflowYProp
}

export interface OverflowYStyle extends T.Style {
  overflowY?: OverflowYValue
}

export const overflowY = style<OverflowYProps, OverflowYStyle>({
  propsKeys: ["overflowY", "ofy"]
})

// Visibility

export type VisibilityValue = C.VisibilityProperty

export type VisibilityProp = T.Prop<VisibilityValue>

export interface VisibilityProps extends T.ThemeProps {
  visibility?: VisibilityProp
  vis?: VisibilityProp
}

export interface VisibilityStyle extends T.Style {
  visibility?: VisibilityValue
}

export const visibility = style<VisibilityProps, VisibilityStyle>({
  propsKeys: ["visibility", "vis"]
})

// Overflow Set

export type OverflowSetProps = OverflowProps & OverflowXProps & OverflowYProps

export type OverflowSetStyle = OverflowStyle & OverflowXStyle & OverflowYStyle

export const overflowSet = compose<OverflowSetProps, OverflowSetStyle>([
  overflow,
  overflowX,
  overflowY
])

// Display Set

export type DisplaySetProps = DisplayProps &
  OpacityProps &
  OverflowSetProps &
  VisibilityProps

export type DisplaySetStyle = DisplayStyle &
  OpacityStyle &
  OverflowSetStyle &
  VisibilityStyle

export const displaySet = compose<DisplaySetProps, DisplaySetStyle>([
  display,
  opacity,
  overflowSet,
  visibility
])
