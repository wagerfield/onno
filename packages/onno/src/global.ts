import { compose } from "./style"
import { transition, TransitionProps, TransitionStyle } from "./animation"
import { borderSet, BorderSetProps, BorderSetStyle } from "./border"
import { boxShadow, BoxShadowProps, BoxShadowStyle } from "./shadow"
import { colorSet, ColorSetProps, ColorSetStyle } from "./color"
import { spaceSet, SpaceSetProps, SpaceSetStyle } from "./space"
import { sizeSet, SizeSetProps, SizeSetStyle } from "./layout"
import { textSet, TextSetProps, TextSetStyle } from "./text"

// Global Set

export type GlobalSetProps = BorderSetProps &
  BoxShadowProps &
  ColorSetProps &
  SpaceSetProps &
  SizeSetProps &
  TextSetProps &
  TransitionProps

export type GlobalSetStyle = BorderSetStyle &
  BoxShadowStyle &
  ColorSetStyle &
  SpaceSetStyle &
  SizeSetStyle &
  TextSetStyle &
  TransitionStyle

export const globalSet = compose<GlobalSetProps, GlobalSetStyle>([
  borderSet,
  boxShadow,
  colorSet,
  spaceSet,
  sizeSet,
  textSet,
  transition
])
