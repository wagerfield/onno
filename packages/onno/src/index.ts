// Core
export * from "./types"
export * from "./constants"
export * from "./utils"
export * from "./compose"
export * from "./style"

// Styles
export * from "./align"
export * from "./animation"
export * from "./background"
export * from "./border"
export * from "./color"
export * from "./display"
export * from "./flex"
export * from "./grid"
export * from "./layout"
export * from "./shadow"
export * from "./space"
export * from "./text"
export * from "./transform"
export * from "./global"
export * from "./variants"
