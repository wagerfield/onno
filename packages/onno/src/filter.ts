import * as T from "./types"
import { isArray, push, isNil } from "./utils"
import { compose } from "./compose"

export function filter<P extends T.Props>(
  options: T.FilterOptions<P>
): T.FilterFunction<P> {
  const { propsKeys, renderers, reducer, initial } = options
  const keys: T.Keys = []

  // Add options propsKeys
  if (isArray(propsKeys)) {
    push.apply(keys, propsKeys)
  }

  // Add renderers propsKeys
  if (isArray(renderers)) {
    const filterSet = compose({ name: "filter", renderers })
    push.apply(keys, filterSet.options.propsKeys)
  }

  // Scoped filterProps function
  const filterProps: T.PickFunction<P> = (props: P) => {
    return keys.reduce((acc, key) => {
      return reducer(acc, key, props)
    }, initial(props))
  }

  // Return filterProps function
  return filterProps
}

export function omit<P extends T.Props>(
  options: T.OmitOptions<P>
): T.OmitFunction<P> {
  const propsKeys = options.propsKeys || ["theme"]
  if (!propsKeys.includes("theme")) propsKeys.push("theme")
  return filter({
    propsKeys,
    renderers: options.renderers,
    initial: (props) => Object.assign({}, props),
    reducer: (acc, key) => {
      delete acc[key]
      return acc
    }
  })
}

export function pick<P extends T.Props>(
  options: T.PickOptions<P>
): T.PickFunction<P> {
  return filter({
    propsKeys: options.propsKeys,
    renderers: options.renderers,
    initial: () => ({}),
    reducer: (acc, key, props) => {
      const value = props[key]
      if (!isNil(value)) acc[key] = value
      return acc
    }
  })
}
