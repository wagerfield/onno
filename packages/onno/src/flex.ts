import * as C from "csstype"
import * as T from "./types"
import { style } from "./style"
import { compose } from "./compose"
import { display, DisplayProps, DisplayStyle } from "./display"
import {
  alignContent,
  AlignContentProps,
  AlignContentStyle,
  alignItems,
  AlignItemsProps,
  AlignItemsStyle,
  alignSelf,
  AlignSelfProps,
  AlignSelfStyle,
  justifyContent,
  JustifyContentProps,
  JustifyContentStyle,
  justifyItems,
  JustifyItemsProps,
  JustifyItemsStyle,
  justifySelf,
  JustifySelfProps,
  JustifySelfStyle,
  order,
  OrderProps,
  OrderStyle
} from "./align"

// Flex

export type FlexValue = C.FlexProperty<T.Length>

export type FlexProp = T.Prop<FlexValue>

export interface FlexProps extends T.ThemeProps {
  flex?: FlexProp
  fx?: FlexProp
}

export interface FlexStyle extends T.Style {
  flex?: FlexValue
}

export const flex = style<FlexProps, FlexStyle>({
  propsKeys: ["flex", "fx"]
})

// Flex Basis

export type FlexBasisValue = C.FlexBasisProperty<T.Length>

export type FlexBasisProp = T.Prop<FlexBasisValue>

export interface FlexBasisProps extends T.ThemeProps {
  flexBasis?: FlexBasisProp
  fxb?: FlexBasisProp
}

export interface FlexBasisStyle extends T.Style {
  flexBasis?: FlexBasisValue
}

export const flexBasis = style<FlexBasisProps, FlexBasisStyle>({
  propsKeys: ["flexBasis", "fxb"]
})

// Flex Grow

export type FlexGrowValue = C.Globals | number

export type FlexGrowProp = T.Prop<FlexGrowValue>

export interface FlexGrowProps extends T.ThemeProps {
  flexGrow?: FlexGrowProp
  fxg?: FlexGrowProp
}

export interface FlexGrowStyle extends T.Style {
  flexGrow?: FlexGrowValue
}

export const flexGrow = style<FlexGrowProps, FlexGrowStyle>({
  propsKeys: ["flexGrow", "fxg"]
})

// Flex Shrink

export type FlexShrinkValue = C.Globals | number

export type FlexShrinkProp = T.Prop<FlexShrinkValue>

export interface FlexShrinkProps extends T.ThemeProps {
  flexShrink?: FlexShrinkProp
  fxs?: FlexShrinkProp
}

export interface FlexShrinkStyle extends T.Style {
  flexShrink?: FlexShrinkValue
}

export const flexShrink = style<FlexShrinkProps, FlexShrinkStyle>({
  propsKeys: ["flexShrink", "fxs"]
})

// Flex Flow

export type FlexFlowValue = C.FlexFlowProperty

export type FlexFlowProp = T.Prop<FlexFlowValue>

export interface FlexFlowProps extends T.ThemeProps {
  flexFlow?: FlexFlowProp
  fxf?: FlexFlowProp
}

export interface FlexFlowStyle extends T.Style {
  flexFlow?: FlexFlowValue
}

export const flexFlow = style<FlexFlowProps, FlexFlowStyle>({
  propsKeys: ["flexFlow", "fxf"]
})

// Flex Direction

export type FlexDirectionValue = C.FlexDirectionProperty

export type FlexDirectionProp = T.Prop<FlexDirectionValue>

export interface FlexDirectionProps extends T.ThemeProps {
  flexDirection?: FlexDirectionProp
  fxd?: FlexDirectionProp
}

export interface FlexDirectionStyle extends T.Style {
  flexDirection?: FlexDirectionValue
}

export const flexDirection = style<FlexDirectionProps, FlexDirectionStyle>({
  propsKeys: ["flexDirection", "fxd"]
})

// Flex Wrap

export type FlexWrapValue = C.FlexWrapProperty

export type FlexWrapProp = T.Prop<FlexWrapValue>

export interface FlexWrapProps extends T.ThemeProps {
  flexWrap?: FlexWrapProp
  fxw?: FlexWrapProp
}

export interface FlexWrapStyle extends T.Style {
  flexWrap?: FlexWrapValue
}

export const flexWrap = style<FlexWrapProps, FlexWrapStyle>({
  propsKeys: ["flexWrap", "fxw"]
})

// Flex Parent Set

export type FlexParentSetProps = DisplayProps &
  AlignItemsProps &
  AlignContentProps &
  JustifyItemsProps &
  JustifyContentProps &
  FlexFlowProps &
  FlexDirectionProps &
  FlexWrapProps

export type FlexParentSetStyle = DisplayStyle &
  AlignItemsStyle &
  AlignContentStyle &
  JustifyItemsStyle &
  JustifyContentStyle &
  FlexFlowStyle &
  FlexDirectionStyle &
  FlexWrapStyle

export const flexParentSet = compose<FlexParentSetProps, FlexParentSetStyle>({
  name: "flexParent",
  renderers: [
    display,
    alignItems,
    alignContent,
    justifyItems,
    justifyContent,
    flexFlow,
    flexDirection,
    flexWrap
  ]
})

// Flex Child Set

export type FlexChildSetProps = AlignSelfProps &
  JustifySelfProps &
  OrderProps &
  FlexProps &
  FlexBasisProps &
  FlexGrowProps &
  FlexShrinkProps

export type FlexChildSetStyle = AlignSelfStyle &
  JustifySelfStyle &
  OrderStyle &
  FlexStyle &
  FlexBasisStyle &
  FlexGrowStyle &
  FlexShrinkStyle

export const flexChildSet = compose<FlexChildSetProps, FlexChildSetStyle>({
  name: "flexChild",
  renderers: [
    alignSelf,
    justifySelf,
    order,
    flex,
    flexBasis,
    flexGrow,
    flexShrink
  ]
})

// Flex Set

export type FlexSetProps = FlexParentSetProps & FlexChildSetProps

export type FlexSetStyle = FlexParentSetStyle & FlexChildSetStyle

export const flexSet = compose<FlexSetProps, FlexSetStyle>({
  name: "flex",
  renderers: [flexParentSet, flexChildSet]
})
