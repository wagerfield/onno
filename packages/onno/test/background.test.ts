import * as O from "../src"
import * as U from "./test-utils"

test("backgroundImage", () => {
  const testProps = U.snapshot(O.backgroundImage)
  testProps({ backgroundImage: "src(foo.png)" })
  testProps({ bgi: "src(bar.png)" })
})

test("backgroundPosition", () => {
  const testProps = U.snapshot(O.backgroundPosition)
  testProps({ backgroundPosition: "center" })
  testProps({ bgp: "0 50%" })
  testProps({ bgp: 100 })
  testProps({ bgp: 0.2 })
  testProps({ bgp: 0 })
  testProps({ bgp: 1 })
})

test("backgroundRepeat", () => {
  const testProps = U.snapshot(O.backgroundRepeat)
  testProps({ backgroundRepeat: "none" })
  testProps({ bgr: "repeat-x" })
})

test("backgroundSize", () => {
  const testProps = U.snapshot(O.backgroundSize)
  testProps({ backgroundSize: "cover" })
  testProps({ backgroundSize: 3 / 4 })
  testProps({ backgroundSize: 0 })
  testProps({ backgroundSize: 1 })
  testProps({ bgs: "contain" })
  testProps({ bgs: 20 })
})

test("backgroundSet", () => {
  const testProps = U.snapshot(O.backgroundSet)
  testProps({
    background: "red url(baz.jpg) center no-repeat",
    backgroundColor: "link",
    backgroundImage: "src(baz.png)",
    backgroundPosition: "left bottom",
    backgroundRepeat: "repeat-y",
    backgroundSize: "cover"
  })
  testProps({
    bg: "text",
    bgc: "gray.1",
    bgi: "src(zoo.png)",
    bgp: "top right",
    bgr: "repeat-x",
    bgs: "contain"
  })
})
