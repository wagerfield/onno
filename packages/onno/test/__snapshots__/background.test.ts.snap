// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`backgroundImage: {
  "backgroundImage": "src(foo.png)"
} 1`] = `
Array [
  Object {
    "backgroundImage": "src(foo.png)",
  },
]
`;

exports[`backgroundImage: {
  "bgi": "src(bar.png)"
} 1`] = `
Array [
  Object {
    "backgroundImage": "src(bar.png)",
  },
]
`;

exports[`backgroundPosition: {
  "backgroundPosition": "center"
} 1`] = `
Array [
  Object {
    "backgroundPosition": "center",
  },
]
`;

exports[`backgroundPosition: {
  "bgp": "0 50%"
} 1`] = `
Array [
  Object {
    "backgroundPosition": "0 50%",
  },
]
`;

exports[`backgroundPosition: {
  "bgp": 0
} 1`] = `
Array [
  Object {
    "backgroundPosition": 0,
  },
]
`;

exports[`backgroundPosition: {
  "bgp": 0.2
} 1`] = `
Array [
  Object {
    "backgroundPosition": "20%",
  },
]
`;

exports[`backgroundPosition: {
  "bgp": 1
} 1`] = `
Array [
  Object {
    "backgroundPosition": "100%",
  },
]
`;

exports[`backgroundPosition: {
  "bgp": 100
} 1`] = `
Array [
  Object {
    "backgroundPosition": "100px",
  },
]
`;

exports[`backgroundRepeat: {
  "backgroundRepeat": "none"
} 1`] = `
Array [
  Object {
    "backgroundRepeat": "none",
  },
]
`;

exports[`backgroundRepeat: {
  "bgr": "repeat-x"
} 1`] = `
Array [
  Object {
    "backgroundRepeat": "repeat-x",
  },
]
`;

exports[`backgroundSet: {
  "background": "red url(baz.jpg) center no-repeat",
  "backgroundColor": "link",
  "backgroundImage": "src(baz.png)",
  "backgroundPosition": "left bottom",
  "backgroundRepeat": "repeat-y",
  "backgroundSize": "cover"
} 1`] = `
Array [
  Object {
    "background": "red url(baz.jpg) center no-repeat",
  },
  Object {
    "backgroundColor": "#00F",
  },
  Object {
    "backgroundImage": "src(baz.png)",
  },
  Object {
    "backgroundPosition": "left bottom",
  },
  Object {
    "backgroundRepeat": "repeat-y",
  },
  Object {
    "backgroundSize": "cover",
  },
]
`;

exports[`backgroundSet: {
  "bg": "text",
  "bgc": "gray.1",
  "bgi": "src(zoo.png)",
  "bgp": "top right",
  "bgr": "repeat-x",
  "bgs": "contain"
} 1`] = `
Array [
  Object {
    "background": "#222",
  },
  Object {
    "backgroundColor": "#AAA",
  },
  Object {
    "backgroundImage": "src(zoo.png)",
  },
  Object {
    "backgroundPosition": "top right",
  },
  Object {
    "backgroundRepeat": "repeat-x",
  },
  Object {
    "backgroundSize": "contain",
  },
]
`;

exports[`backgroundSize: {
  "backgroundSize": "cover"
} 1`] = `
Array [
  Object {
    "backgroundSize": "cover",
  },
]
`;

exports[`backgroundSize: {
  "backgroundSize": 0
} 1`] = `
Array [
  Object {
    "backgroundSize": 0,
  },
]
`;

exports[`backgroundSize: {
  "backgroundSize": 0.75
} 1`] = `
Array [
  Object {
    "backgroundSize": "75%",
  },
]
`;

exports[`backgroundSize: {
  "backgroundSize": 1
} 1`] = `
Array [
  Object {
    "backgroundSize": "100%",
  },
]
`;

exports[`backgroundSize: {
  "bgs": "contain"
} 1`] = `
Array [
  Object {
    "backgroundSize": "contain",
  },
]
`;

exports[`backgroundSize: {
  "bgs": 20
} 1`] = `
Array [
  Object {
    "backgroundSize": "20px",
  },
]
`;
